src/openfermion/ops/_polynomial_tensor.py

```diff
@@ -250,26 +250,47 @@ def __sub__(self, subtrahend):
         return r
 
     def __imul__(self, multiplier):
-        if not issubclass(type(multiplier), PolynomialTensor):
+        if isinstance(multiplier, (int, float, complex)):
+            for key in self.n_body_tensors:
+                self.n_body_tensors[key] *= multiplier
+
+        elif isinstance(multiplier, PolynomialTensor):
```

Comment:
ah, thanks for this.

Reply:
No problem, I should have caught it the first time around.